Documentation/design-docs/host-error-codes.md

```diff
@@ -0,0 +1,110 @@
+
+* `CoreHostLibMissingFailure` (`0x80008083`) - One of the dependent libraries is missing. Typically when the `hostfxr`, `hostpolicy` or `coreclr` dynamic libraries are not present in the expected locations. Probably means corrupted or incomplete installation.
+
+* `CoreHostEntryPointFailure` (`0x80008084`) - One of the dependent libraries is missing a required entry point. For example if 3.0 `hostfxr` is used via the `initialize_for_runtime_config` and the framework resolution logic ends up loading 2.0 `hostpolicy` which doesn't support the new behaviors.
+
+* `CoreHostCurHostFindFailure` (`0x80008085`) - If the hosting component is trying to use the path to the current module (the hosting component itself) and from it deduce the location of the installation. Either the location of the current module could not be determined (some weird OS call failure) or the location is not in the right place relative to other expected components.  
+For example the `hostfxr` may look at its location and try to deduce the location of the `shared` folder with the framework from it. It assumes the typical install layout on disk. If this doesn't work, this error will be returned.
+
+* `CoreClrResolveFailure` (`0x80008087`) - If the `coreclr` could not be found. The hosting layer (`hostpolicy`) looks for `coreclr` library either in the app itself (for self-contained) of in the root framework (for framework-dependent). This search can be done purely by looking at disk or more commonly by looking into the respective `.deps.json`. If `coreclr` is missing in `.deps.json` or it's there but doesn't exist on disk, this error is returned.
+
+* `CoreClrBindFailure` (`0x80008088`) - The loaded `coreclr` doesn't have one of the required entry points.
+
+* `CoreClrInitFailure` (`0x80008089`) - The call to `coreclr_initialize` failed. The actual error returned by `coreclr` is reported in the error message.
+
+* `CoreClrExeFailure` (`0x8000808a`) - The call to `coreclr_execute_assembly` failed. Note that this does not mean anything about the app's exit code, this failure occurs if `coreclr` failed to run the app itself.
+
+* `ResolverInitFailure` (`0x8000808b`) - Initialization of the `hostpolicy` dependency resolver failed. This can be:
+  * One of the frameworks or the app is missing a required `.deps.json` file.
+  * One of the `.deps.json` files is invalid (invalid JSON, or missing required properties and so on).
+
+* `ResolverResolveFailure` (`0x8000808c`) - Resolution of dependencies in the `hostpolicy` dependency resolved failed. This can mean many different things, but in general one of the processed `.deps.json` contains entry for a file which could not found, or its resolution failed for some other reason (conflict for example).
```

Comment:
```suggestion
* `ResolverResolveFailure` (`0x8000808c`) - Resolution of dependencies in `hostpolicy` failed. This can mean many different things, but in general one of the processed `.deps.json` contains entry for a file which could not found, or its resolution failed for some other reason (conflict for example).
```